docs/user_guide/command_line_interface.rst

```diff
@@ -31,45 +31,48 @@ The available options:
 
 ::
 
- usage: preprocess [-h] [--files files [files ...]] [--dataset dataset] [--num_partitions num_partitions] [--overwrite]
- [--generate_config [generate_config]] [--format format] [--delim delim] [--dtype dtype] [--not_remap_ids]
- [--dataset_split dataset_split dataset_split] [--start_col start_col] [--num_line_skip num_line_skip]
- output_directory
-
- Preprocess Datasets
-
- positional arguments:
- output_directory      Directory to put graph data
-
- optional arguments:
- -h, --help            show this help message and exit
- --files files [files ...]
-     Files containing custom dataset
- --dataset dataset     Supported dataset to preprocess
- --num_partitions num_partitions
-     Number of partitions to split the edges into
- --overwrite           Overwrites the output_directory if this issetOtherwise, files with same the names will be treated as the data for current dataset.
- --generate_config [generate_config], -gc [generate_config]
-     Generates a single-GPU/multi-CPU/multi-GPU training configuration file by default.
-     Valid options (default to GPU): [GPU, CPU, multi-GPU]
- --format format       Format of data, eg. srd
- --delim delim, -d delim
-     Specifies the delimiter
- --dtype dtype         Indicates the numpy.dtype
- --not_remap_ids       If set, will not remap ids
- --dataset_split dataset_split dataset_split, -ds dataset_split dataset_split
-     Split dataset into specified fractions
- --start_col start_col, -sc start_col
-     Indicates the column index to start from
- --num_line_skip num_line_skip, -nls num_line_skip
-     Indicates number of lines to skip from the beginning
-
- Specify certain config (optional): [--<section>.<key>=<value>]
+    usage: preprocess [-h] [--download_directory download_directory] [--files files [files ...] | --dataset dataset] [--num_partitions num_partitions]
+                    [--overwrite] [--generate_config [generate_config]] [--format format] [--delim delim] [--dtype dtype] [--not_remap_ids]
```

Comment:
do all these inputs have default values? if so where are these values set? also what is the flag `dtype`?

Reply:
| flag           | default value     | location of these values
| ----------- | ----------------- | --------------------------------------------------------------------------|
| overwrite  | False                  | if set, the corresponding behavior is conducted, mentioned in ``command_line_interface.rst line 54``
| generate_config| False         | if set, the corresponding behavior is conducted, mentioned in ``command_line_interface.rst line 56``
| format | "srd" | Set in ``preprocess line 1217``, didn't mentioned in docs
| delim  | Use Python Sniffer to detect | mentioned in ``command_line_interface.rst line 198``
| dtype | ``int32`` | mentioned in ``command_line_interface.rst line 212``
| not_remap_ids| False | if set, the corresponding behavior is conducted, mentioned in ``command_line_interface.rst line 62``

Reply:
``--dtype <dtype>`` is an **optional** argument for ``marius_preprocess``.
It defines the format for storing each node remapped ID and relation remapped ID. The current supported
format is ``int32`` and ``int64``. 
When storing in ``int32``, each remapped ID will be a 4-byte integer.
When storing in ``int64``, each remapped ID will be a 8-byte integer.
If the total number of nodes or relations is smaller than 2.1 billion,
it is recommended to choose ``int32`` to avoid unnecessary waste of space.
On the other hand, if the total number of nodes or relations is bigger than 2.1 billion,
it is recommended to choose``int64`` to ensure all IDs to be remapped.
The default ``<dtype>`` is set to ``int32``.

This definition is in ``command_line_interface.rst line 203-212``